### PR TITLE
Add migration guide for RESTEasy Reactive

### LINKED ISSUE #24239: Write a RESTEasy Classic -> Reactive migration guide
We need to collect all the most common use cases and document them.

We should also document the limitations (typically the lack of GZip support if we don't have it by then).

https://github.com/quarkusio/quarkus-quickstarts/pull/971 is a good start for that.

### PR DESCRIPTION
**Pull request by geoand**
Closes: #24239

### CODE REVIEW

**Comment by geoand** [comment]
@FroMage @stuartwdouglas if you have any thoughts on this as I have probably forgot various things.

@michalszynkiewicz same for the client part :).

Feel free to update the PR if you like.

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 1]
```diff
@@ -0,0 +1,152 @@
+////
+This file is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Migrating to RESTEasy Reactive
+
+Migrating from the classic RESTEasy to RESTEasy Reactive is straightforward in most cases, however there are a few cases that require some attention.
```
Comment (gsmet) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:8]:
```suggestion
Migrating from RESTEasy Classic to RESTEasy Reactive is straightforward in most cases, however there are a few cases that require some attention.
```

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+|`org.jboss.resteasy.annotations.SseElementType`
+|`org.jboss.resteasy.reactive.RestStreamElementType`
+|
+
+|===
+
+=== JAX-RS providers
+
+Although RESTEasy Reactive provides the same spec compliant behavior as classic RESTEasy does, it does not include the same exact provider implementations at runtime.
```
Comment (gsmet) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:93]:
```suggestion
Although RESTEasy Reactive provides the same spec compliant behavior as RESTEasy Classic does, it does not include the same exact provider implementations at runtime.
```

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+The most common case where the difference in providers might result in different behavior, is the included `javax.ws.rs.ext.ExceptionMapper` implementations. To see what classes are included in the application, launch the application in dev mode and navigate to http://localhost:8080/q/dev/io.quarkus.quarkus-resteasy-reactive/exception-mappers.
+
+=== Multipart support
+
+HTTP Multipart support in RESTEasy Reactive does **not** reuse the same types or annotations as classic RESTEasy and thus users are encouraged to read <<resteasy-reactive#multipart,this>> part of the reference documentation.
```
Comment (gsmet) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:99]:
```suggestion
HTTP Multipart support in RESTEasy Reactive does **not** reuse the same types or annotations as RESTEasy Classic and thus users are encouraged to read <<resteasy-reactive#multipart,this>> part of the reference documentation.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Given we went with the "brand", better update it in there too.

**Review by gsmet — APPROVED** [review]
Let's get this in, we can still expand later if needed.

Thanks!

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+Quarkus uses smart defaults when determining the media type of JAX-RS methods in order to simplify common use cases.
+The difference between `quarkus-resteasy-reactive` and `quarkus-resteasy` is the use of `text/plain` as the default media type instead of `text/html`
+when the method returns a `String`.
+
+== Client
+
+The Reactive REST Client (`quarkus-rest-client-reactive` and its dependencies) replace the legacy `quarkus-rest-client` but leverage Quarkus' build time processing
+and the unified I/O model provided by Vert.x.
+
```
Comment (michalszynkiewicz) [inline docs/src/main/asciidoc/resteasy-reactive-migration.adoc:111]: maybe https://quarkus.io/guides/rest-client-reactive#known-limitations is worth linking here too

**Review by michalszynkiewicz — APPROVED** [review]

File: docs/src/main/asciidoc/resteasy-reactive-migration.adoc (hunk 4 above)
Comment (geoand) [reply docs/src/main/asciidoc/resteasy-reactive-migration.adoc:111]: Feel free to add it :)

**Pull request merged**